### ISSUE TITLE
Why does DI(DI(prod))(v::Vector) produce a static array?

### ISSUE THREAD

**Issue by MasonProtter** [YingboMa/ForwardDiff2.jl#23]
```julia
 v = rand(3)

julia> DI(DI(prod))(v)
 3×3 StaticArrays.SArray{Tuple{3,3},Float64,2,9} with indices SOneTo(3)×SOneTo(3):
  0.0       0.673959  0.940515
  0.673959  0.0       0.768448
  0.940515  0.768448  0.0   
```
This seems a little surprising and suspicious.

**Comment by YingboMa** [comment]
Yes, we know about this. We don't have the ability to select the chunk size yet.

**Comment by MasonProtter** [comment]
Can you explain what the relation between chunk sizing and the output type is?

If the output type of the Hessian is determined by the length of `v`, this will be type unstable.

**Comment by shashi** [comment]
This thing is actually a result of the fact that `getindex` on a `DualArray` returns a `StaticVector`. We started off with making this be a `view` into the underlying partials array if `v` is mutable, but ran into lots of trouble with the compiler.

> Can you explain what the relation between chunk sizing and the output type is?

I think in this case it would still be an array. Any relationship is artificial at the moment. We didn't want to fix this by making it return an Array which is doing more work to return a worse array...
